include/rocksdb/file_system.h

```diff
@@ -108,6 +108,20 @@ struct IOOptions {
   // such as NewRandomAccessFile and NewWritableFile.
   std::unordered_map<std::string, std::string> property_bag;
 
+  // Sets the property in property_bag.
+  void SetProperty(const std::string& property, const std::string& value) {
```

Comment:
performance recommendation: Because `value` is always copied here, use `std::string&& value` and `= std::move(value);` instead for move semantics.